sdk/core/Azure.Core/src/Serialization/ObjectSerializer.cs

```diff
@@ -31,6 +32,42 @@ public abstract class ObjectSerializer
         /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use during serialization.</param>
         public abstract ValueTask SerializeAsync(Stream stream, object? value, Type inputType, CancellationToken cancellationToken);
 
+        /// <summary>
+        /// Convert the provided value to it's binary representation and return it as a <see cref="BinaryData"/> instance.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        /// <param name="inputType">The type of the <paramref name="value"/> to convert.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use during serialization.</param>
+        /// <returns>The object's binary representation as <see cref="BinaryData"/>.</returns>
+        public BinaryData SerializeToBinaryData(object? value, Type? inputType = default, CancellationToken cancellationToken = default) =>
```

Comment:
is "inputType" the naming convention we use? BTW, I think the docs for "inputType" should be better. I don't think anybody will understand what this parameter is for based on the current description.

Reply:
This is the name that is used on the Serialize/Deserialize methods of ObjectSerializer. I can improve the docs.